modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotDeletionExecutor.java

```diff
@@ -115,7 +115,13 @@ public void run() {
                             } else {
                                 SlotDeliveryWorker slotWorker = SlotDeliveryWorkerManager.getInstance()
                                         .getSlotWorker(slot.getStorageQueueName());
-                                slotWorker.deleteSlot(slot);
+
+                                // slotWorker can be null.
+                                // For instance if the deletion request came from coordinator after a member leaves the
+                                // cluster and no subscribers on coordinator node.
+                                if (null != slotWorker) {
+                                    slotWorker.deleteSlot(slot);
```

Comment:
I have stated the scenario in the comment above. (with this fix coordinator will schedule to delete all the empty slots when a member leaves. If there are no subscribers at coordinator node, slotWorker can be null)